connection-loader/src/main/java/com/terracotta/diagnostic/DiagnosticEntityClientService.java

```diff
@@ -52,15 +53,27 @@ public Diagnostics create(final EntityClientEndpoint<EntityMessage, EntityRespon
     return (Diagnostics)Proxy.newProxyInstance(getClass().getClassLoader(), new Class[] {Diagnostics.class},
             new java.lang.reflect.InvocationHandler() {
       @Override
-      public Object invoke(Object proxy, final Method method, Object[] args) throws Throwable {
+      public Object invoke(Object proxy, Method method, final Object[] args) throws Throwable {
         try {
-          return ece.beginInvoke().message(new EntityMessage() {
+          final String methodName = method.getName();
+          InvokeFuture returnValue = ece.beginInvoke().message(new EntityMessage() {
             @Override
             public String toString() {
-              return method.getName();
+              if (methodName.equals("get")) {
+                return "getJMX " + args[0] + " " + args[1];
+              } else if (methodName.equals("set")) {
+                return "setJMX " + args[0] + " " + args[1] + " " + args[2];
+              } else if (methodName.equals("invoke")) {
+                return "invokeJMX " + args[0] + " " + args[1];
+              } else {
+                return methodName;
+              }
             }
-
-          }).invoke().get().toString();
+          }).invoke();
+          // if the server is terminating, never going to get a message back.  just return null
+          if (!methodName.equals("terminateServer") && !methodName.equals("forceTerminateServer")) {
+            return returnValue.get().toString();
```

Comment:
@myronkscott : in a traditional connection, if for example we were to do a call to `getConfig()` and the active crashed, then the call will block.

If we connect with diagnostic and we do a call to a method of this entity, but the server crash or something wrong happens during this time.

What will be the behavior ? The call will block, until it can be fulfilled ?